hs/src/Reach/Eval/Core.hs

```diff
@@ -3934,8 +3934,9 @@ evalStmt = \case
       Just ty
         | ty /= dv_ty -> expect_ $ Err_Try_Type_Mismatch ty dv_ty
         | otherwise   -> return ()
-    saveLift =<< withAt (`DLS_Throw` dv)
+    saveLift =<< withAt (\at -> DLS_Throw at dv $ not $ isConsensusStep $ st_mode curSt)
```

Comment:
I believe that we are interested in the state of the `try`, not the state of the `throw`, because the `try` is essentially going to turn into an `if`